tools/include/eosio/codegen.hpp

```diff
@@ -273,6 +340,26 @@ namespace eosio { namespace cdt {
                cg.notify_handlers.insert(full_notify_name); // insert the method action, so we don't create the dispatcher twice
             }
 
+            // We allow a method to be tagged as both `action` and `call`
+            if (decl->isEosioCall()) {
+               name = generation_utils::get_call_name(decl);
+               validate_name(name, [&](auto s) {
+                  CDT_ERROR("codegen_error", decl->getLocation(), std::string("call name (")+s+") is not a valid eosio name");
+               });
+
+               if (!_call_set.count(name))
+                  _call_set.insert(name);
+               else {
+                  auto itr = _call_set.find(name);
+                  CDT_CHECK_ERROR(*itr == name, "codegen_error", decl->getLocation(), "call declaration doesn't match previous declaration");
+               }
+               std::string full_call_name = decl->getNameAsString() + ((decl->getParent()) ? decl->getParent()->getNameAsString() : "");
+               if (cg.calls.count(full_call_name) == 0) {
+                  create_call_dispatch(decl);
+               }
+               cg.calls.insert(full_call_name); // insert the method call, so we don't create the dispatcher twice
```

Comment:
Why no:
```c++
               if (decl->isEosioReadOnly()) {
                  read_only_actions.insert(decl);
               }
```

Reply:
`read_only` handling in CDT is not completely working. It is going to be repurposed as an only indication. I did not want to add anything new with it.